docker/auth/auth.py

```diff
@@ -105,6 +114,34 @@ def resolve_authconfig(authconfig, registry=None):
     return None
 
 
+def _resolve_authconfig_credstore(authconfig, registry, credstore_name):
+    if not registry or registry == INDEX_NAME:
+        # The ecosystem is a little schizophrenic with index.docker.io VS
+        # docker.io - in that case, it seems the full URL is necessary.
+        registry = 'https://index.docker.io/v1/'
+    log.debug("Looking for auth entry for {0}".format(repr(registry)))
+    if registry not in authconfig:
+        log.debug("No entry found")
+    store = dockerpycreds.Store(credstore_name)
+    try:
+        data = store.get(registry)
+        res = {
+            'ServerAddress': registry,
+        }
+        if data['Username'] == TOKEN_USERNAME:
+            res['IdentityToken'] = data['Secret']
+        else:
+            res.update({
+                'Username': data['Username'],
+                'Password': data['Secret'],
+            })
+        return res
+    except dockerpycreds.StoreError as e:
+        log.error('Credentials store error: {0}'.format(repr(e)))
+
```

Comment:
I wonder if this should raise an exception? The user explicitly asked to use the credential store, right ?